pkg/syncer/metadatasyncer.go

```diff
@@ -970,7 +970,6 @@ func csiPVUpdated(ctx context.Context, newPv *v1.PersistentVolume, oldPv *v1.Per
 // csiPVDeleted deletes volume metadata on VC when volume has been deleted on Vanills k8s and supervisor cluster
 func csiPVDeleted(ctx context.Context, pv *v1.PersistentVolume, metadataSyncer *metadataSyncInformer) {
 	log := logger.GetLogger(ctx)
-	var deleteDisk bool
 	if pv.Spec.ClaimRef != nil && (pv.Status.Phase == v1.VolumeAvailable || pv.Status.Phase == v1.VolumeReleased) && pv.Spec.PersistentVolumeReclaimPolicy == v1.PersistentVolumeReclaimDelete {
```

Comment:
Can we correct this condition?
It should be as following. The below state represents a deletion of PVC with reclaim policy delete set on the PV. The controller will help delete the volume for this case.

```
 	if pv.Spec.ClaimRef != nil && pv.Status.Phase == v1.VolumeReleased && pv.Spec.PersistentVolumeReclaimPolicy == v1.PersistentVolumeReclaimDelete {
```

Reply:
As discussed I have addressed the changes